core/keyboard_nav/navigation.js

```diff
@@ -302,8 +302,12 @@ Blockly.navigation.insertFromFlyout = function() {
     return;
   }
 
-  var newBlock = flyout.createBlock(
-      Blockly.navigation.getFlyoutCursor_().getCurNode().getLocation());
+  var curBlock = Blockly.navigation.getFlyoutCursor_().getCurNode().getLocation();
+  if (!curBlock.isEnabled()) {
+    return;
```

Comment:
Can you add a warning message here? Something like 'Can't insert disabled block'

Reply:
Yep, done.